src/qt/guiutil.cpp

```diff
@@ -499,8 +598,15 @@ bool SetStartOnSystemStartup(bool fAutoStart)
         // Write a bitcoin.desktop file to the autostart directory:
         optionFile << "[Desktop Entry]\n";
         optionFile << "Type=Application\n";
-        optionFile << "Name=Gridcoin\n";
-        optionFile << "Exec=" << pszExePath << " -min\n";
+        optionFile << "Name=Gridcoin" + autostartup.link_name_suffix + "\n";
+        optionFile << "Exec=" << pszExePath;
+
+        if (!autostartup.data_dir.empty())
+        {
+            optionFile << " -datadir=" << autostartup.data_dir;
```

Comment:
```suggestion
            optionFile << " -datadir='" << autostartup.data_dir << "'";
```
Need to shell-quote the `-datadir` argument in case that path has spaces in any of the directory names.

Reply:
Agreed.

Reply:
Did we lose this one in the patch? I don't see the quotes in the commit. 

Nice that GitHub provides a button for that. Here it is again in case it makes it any easier:
```suggestion
            optionFile << " -datadir='" << autostartup.data_dir << "'";
```

Reply:
Oh, I suppose it's because this thread was marked resolved.